lib/rules/color-named/index.js

```diff
@@ -58,11 +58,23 @@ const rule = function (expectation, options) {
         return
       }
 
+      const functionNodeIndexes = []
+
       valueParser(decl.value).walk(node => {
         const value = node.value,
           type = node.type,
           sourceIndex = node.sourceIndex
 
+        // Find nodes that are inside functions
+        if (type === "function") {
+          node.nodes.forEach((node) => functionNodeIndexes.push(node.sourceIndex))
+        }
+
+        // Return early if node is inside of a function
+        if (functionNodeIndexes.indexOf(sourceIndex) > -1) {
```

Comment:
Was just about to mention that - as it appears that this is essentially the same behavior as the `ignore` `inside-function` option. Which makes me ask, what is https://github.com/stylelint/stylelint/issues/2009 asking of that can't be achieved with that option?